src/auto-evo/simulation/MichePopulation.cs

```
@@ -141,157 +163,112 @@ private static List<Species> CopyInitialPopulationsToResults(SimulationConfigura
-        bool dayNightCycle = worldSettings.DayNightCycleEnabled;
 
-        // This algorithm version is for microbe species
-        // TODO: add simulation for multicellular
-        var species = genericSpecies.Select(s => s as MicrobeSpecies).Where(s => s != null).Select(s => s!)
-            .ToList();
+        populations.MicheByPatch.TryGetValue(patch, out var miche);
+
+        foreach (var s in simulationConfiguration.ExtraSpecies)
+            miche!.InsertSpecies(s, cache);
+
+        var species = genericSpecies.Where(x => !simulationConfiguration.ExcludedSpecies.Contains(x))
+            .Union(simulationConfiguration.ExtraSpecies).Distinct().ToList();
 
         // Skip if there aren't any species in this patch
         if (species.Count < 1)
             return;
 
-        var energyBySpecies = new Dictionary<MicrobeSpecies, float>();
-        foreach (var currentSpecies in species)
-        {
-            energyBySpecies[currentSpecies] = 0.0f;
-        }
-
-        bool strictCompetition = autoEvoConfiguration.StrictNicheCompetition;
-
-        var niches = new List<FoodSource>
-        {
-            new EnvironmentalFoodSource(patch, Sunlight, Constants.AUTO_EVO_SUNLIGHT_ENERGY_AMOUNT),
-            new EnvironmentalFoodSource(patch, Temperature, Constants.AUTO_EVO_THERMOSYNTHESIS_ENERGY_AMOUNT),
-            new CompoundFoodSource(patch, Glucose, dayNightCycle),
-            new CompoundFoodSource(patch, HydrogenSulfide, dayNightCycle),
-            new CompoundFoodSource(patch, Iron, dayNightCycle),
-            new ChunkFoodSource(patch, "marineSnow"),
-            new ChunkFoodSource(patch, "ironSmallChunk"),
-            new ChunkFoodSource(patch, "ironBigChunk"),
-        };
-
-        foreach (var currentSpecies in species)
-        {
-            niches.Add(new HeterotrophicFoodSource(patch, currentSpecies, cache));
-        }
+        var leafNodes = miche!.GetLeafNodes().Where(x => x.Occupant != null).ToList();
```

If getting the miche fails above, it should probably throw an exception as something is very wrong.